docs/products/flink/howto/upgrade-flink-version.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Upgrade Aiven for Apache Flink
+---
+
+Upgrading to the latest version of Aiven for Apache Flink® allows you to benefit from improved features, enhanced performance, and better security.
+
+:::note
+Aiven for Apache Flink version 1.16 will soon reach
+[end-of-life](/docs/platform/reference/eol-for-major-versions#aiven-for-flink).
+:::
+
+## Limitations
+
+- Direct upgrades on active Aiven for Apache Flink services are not supported.
+- Upgrading to Apache Flink 1.19 requires creating a new service and manually
+  transferring applications.
+
+## Migrate to a newer Apache Flink version
+
+If you are using Aiven for Apache Flink version 1.16 and want to take advantage of
+the enhancements in Aiven for Apache Flink 1.19, you must migrate your applications
+to a new Flink service. Follow these steps:
+
+### Step 1: Create new Aiven for Apache Flink service
+
+1. Create new Aiven for Apache Flink service using the
+   [Aiven Console](https://console.aiven.io/), [Aiven Client](/docs/tools/cli/service/flink),
+   or [Terraform](/docs/tools/terraform) (provider version 4.19.0).
+1. Select Apache Flink 1.19 as the deployment version for your new
+   Aiven for Apache Flink service.
+
+### Step 2: Transfer your applications
+
+**Manual transfer**: If you created your Aiven for Apache Flink applications manually:
+
+  1. Stop the application deployment in the old Aiven for Apache Flink service.
+  1. Copy the source, sinks, and transformation statements from the old service for your
+     applications.
+  1. Create new applications in the new service and add the copied source, sinks, and
+     transformation statements.
+  1. Deploy the new applications in the new service.
+
+     For more information, see
+    [Aiven for Apache Flink applications](/docs/products/flink/howto/create-flink-applications).
+
+**Using Terraform**: If you used Terraform (ensure you’re on provider version 4.19.0)
+  to create your service and applications:
+
+  1. Update the `flink_version` property in your Terraform script to 1.19.
```

Comment:
Have you @harshini-rangaswamy tried this works? I was thinking this too and I wasn't sure whether this would try to change the `flink_version` of an existing Flink service or create a new one.

Reply:
> Have you @harshini-rangaswamy tried this works? I was thinking this too and I wasn't sure whether this would try to change the `flink_version` of an existing Flink service or create a new one.

No. This is based on the information provided in [HH-3856](https://aiven.atlassian.net/browse/HH-3856). It would be great if you could test this out and validate it. Thank you.

Reply:
Right. I later started to think whether this changing actually works. Needs to be checked!

Reply:
Ah, when I read it my mind just went straight to thinking it will replace the resource.

Reply:
It will not work as it currently is: https://aiven-io.slack.com/archives/C0188EGC3GB/p1717151087913929